engine/src/main/java/org/terasology/entitySystem/entity/EntityBuilder.java

```diff
@@ -159,11 +187,11 @@ public void setAlwaysRelevant(boolean alwaysRelevant) {
     }
 
     public void setScope(EntityScope scope) {
-        this.scope = scope;
+        this.scope = Optional.of(scope);
     }
 
     public EntityScope getScope() {
-        return scope;
+        return scope.orElse(null);
```

Comment:
Since you hold the scope internally as `Optional`, why not expose it? IMHO the internal interfaces looks nicer than the external one here...